packages/binding-coap/src/coap-server.ts

```diff
@@ -182,17 +185,73 @@ export default class CoapServer implements ProtocolServer {
         return `${this.scheme}://${address}:${port}/${encodeURIComponent(urlPath)}`;
     }
 
+    private fillInMetaPropertiesBindingData(thing: ExposedThing, base: string, offeredMediaType: string) {
+        const opValues = this.createPropertyMetaOpValues(thing);
+
+        if (opValues.length === 0) {
+            return;
+        }
+
+        if (thing.forms == null) {
+            thing.forms = [];
+        }
+
+        const form = this.createAffordanceForm(base, this.PROPERTY_DIR, offeredMediaType, opValues, thing.uriVariables);
+
+        thing.forms.push(form);
+    }
+
+    private getReadableProperties(thing: ExposedThing) {
+        return Object.entries(thing.properties).filter(([_, value]) => value.writeOnly !== true);
+    }
+
+    private getWritableProperties(thing: ExposedThing) {
+        return Object.entries(thing.properties).filter(([_, value]) => value.readOnly !== true);
+    }
+
+    private createPropertyMetaOpValues(thing: ExposedThing): string[] {
+        const properties = Object.values(thing.properties);
+        const numberOfProperties = properties.length;
+
+        if (numberOfProperties === 0) {
+            return [];
+        }
+
+        const readableProperties = this.getReadableProperties(thing).length;
+        const writableProperties = this.getWritableProperties(thing).length;
+
+        const opValues: string[] = [];
+
+        if (readableProperties > 0) {
+            opValues.push("readmultipleproperties");
+        }
+
+        if (readableProperties === numberOfProperties) {
+            opValues.push("readallproperties");
+        }
+
+        if (writableProperties > 0) {
+            opValues.push("writemultipleproperties");
+        }
+
+        if (writableProperties === numberOfProperties) {
+            opValues.push("writeallproperties");
+        }
+
+        return opValues;
+    }
+
```

Comment:
Note that these helper functions could also be moved to the core package (later?) to help with achieving unified behavior across binding implementations.